### PR TITLE
added validation

### PR DESCRIPTION
**Pull request by builderjer**

### CODE REVIEW

File: ovos_PHAL_plugin_sj201_led/__init__.py [hunk 1]
```diff
@@ -87,11 +91,23 @@ def turn_off(self):
...
+        # Check for a file created by ovos-i2csound
+        # https://github.com/OpenVoiceOS/ovos-i2csound/blob/a107b74d61bbb362421fb51da6b91f2ba49b6a1a/ovos-i2csound#L76
+        LOG.debug(f"checking file {I2C_PLATFORM_FILE}")
+        if exists(I2C_PLATFORM_FILE):
+            with open(I2C_PLATFORM_FILE) as f:
+                platform = f.readline().strip().lower()
+            if platform == "SJ201LED":
+                return True
+        # Try a direct hardware check
+        if is_mycroft_sj201():
```
Comment (JarbasAl) [inline ovos_PHAL_plugin_sj201_led/__init__.py:107]: has this been validated? not sure this plugin works at all, and if it does i think ONLY in devkit, so we need to exclude rev10 boards

Comment (builderjer) [reply ovos_PHAL_plugin_sj201_led/__init__.py:107]: This is supposed to only allow the dev kit.  I just added validation, have not tested the plugin
reference:
https://github.com/OpenVoiceOS/ovos-PHAL/blob/09f8b33460d2c785623f8b1c0ab73d50762064da/ovos_PHAL/detection.py#L5

But just noticed i2csound needs fixed for each board version.

Comment (builderjer) [reply ovos_PHAL_plugin_sj201_led/__init__.py:107]: ovos-i2csound was fixed [here](https://github.com/OpenVoiceOS/ovos-i2csound/commit/03259b5d4e4521d0189df0d529c0af87be20697b)

With this PR, any auto-detection that I can think of is handled.

**Comment by JarbasAl** [comment]
@builderjer think you can look into #2 and maybe rebase this one on top?

**Comment by builderjer** [comment]
I am merging this, and will add more for the production unit soon.  This at least allows detection for the dev unit.

**Pull request merged**